### ISSUE TITLE
AEM with TOTP: 6 digit code generated; 8 digit code needed 

### ISSUE THREAD

**Issue by S-G-iscoding** [QubesOS/qubes-issues#6494]
**Qubes OS version**

R4.0

**Affected component(s) or functionality**

AEM (Anti Evil Maid)
anti-evil-maid-install

**Brief summary**

While setting up AEM with TOTP (2FA), the installer shows a qr code for setting up a 2FA app. I tried FreeOTP (which is one of the suggested one) as well as Aegis - both are generating 6 digit codes which are not accepted by the installer because it asks for 8 digit code. I had to manually change the settings in the app to 8 digits.

**How Reproducible**

100%

**To Reproduce**

1. Setup AEM with `anti-evil-maid-install` and the `-m` option for 2FA.
2. Scan the QR-Code with FreeOTP or Aegis.
3. Enter the code shown in the app (6 digits)

**Expected behavior**

4. The code is accepted because it is the one generated from the data in the qr code.

**Actual behavior**

4. Qubes says that the code is not correct and it should be another code (8 digits)
5. Change the settings in the app to 8 digits
6. Enter the code
7. Now it is working.

**Screenshots**

**Additional context**

**Solutions you've tried**

Working workaround: Change the settings in Aegis to 8 digits. 

**Relevant [documentation](https://www.qubes-os.org/doc/) you've consulted**

https://www.qubes-os.org/doc/anti-evil-maid/
https://github.com/QubesOS/qubes-antievilmaid/blob/master/README

**Related, [non-duplicate](https://www.qubes-os.org/doc/reporting-bugs/#new-issues-should-not-be-duplicates-of-existing-issues) issues**

**Comment by iamahuman** [comment]
AEM MFA setup should have asked you a generated OTP code for confirmation. Did it require 8 digits from the setup, or only since next boot? In the latter case, please try seeing if the issue persists after reinstalling AEM as well as `oathtool` and the kernel.

**Comment by S-G-iscoding** [comment]
My issue is about the AEM MFA setup which asked me for a 8 digit code for confirmation, but using the qr code the OTP apps are only generating 6 digit codes without further changes in the settings of these apps. I do not know if this issue persists if I boot because at the moment I do not have a booting and not instantly crashing version of AEM (I already made sure I didn't made common errors like using a wrong sinit module etc., but that is another topic).

**Comment by iamahuman** [comment]
This sounds weird, since I cannot yet find anything that indicates usage of 8 digits of code in relevant software.

Are you sure dom0 is up to date and you're using the AEM software from Qubes?

**Comment by rustybird** [comment]
>  4. Qubes says that the code is not correct and it should be another code (8 digits)

If you're referring to a message like:

     oathtool: password "123456" not found in range 53906772 .. 53906772

It may look as if the correct code is 53906772, but the number actually designates a 30-second slice of time that the code 123456 was checked against.

Your computer's clock and your OTP device's clock are probably desynchronized not too much but enough (e.g. 20 seconds) that sometimes the slices of time they fall into coincide, and sometimes they don't. So switching to 8 digits wasn't the reason it succeeded, rather it just made you try again and this time you were lucky. (And `anti-evil-maid-install` unintentionally happens to accept more than 6 digits.)

TL;DR: Sync your clocks :)

**Comment by S-G-iscoding** [comment]
@rustybird: You are right. That "solves" the "issue".

> (And anti-evil-maid-install unintentionally happens to accept more than 6 digits.)

But maybe this could be fixed?
